option.go

```diff
@@ -352,7 +352,7 @@ func (optSet OptionSet) ByFlag(flag string) *Option {
 	}
 	for i := range optSet {
 		opt := &optSet[i]
-		if opt.Flag == flag || opt.FlagShorthand == flag {
```

Comment:
These changes don't need reviewing but it's worth mentioning `pflag` only lets you use the shorthand with a single hyphen, so I'm getting rid of the completion for now.